packages/react/src/components/Popup/Popup.tsx

```diff
@@ -76,19 +81,27 @@ function useRestoreFocus(props: IPopupProps, root: React.RefObject<HTMLDivElemen
       containsFocus.current = true;
     }
 
+    if (nodesToHide) {
+      ariaHidden(nodesToHide, true);
+    }
+
     return () => {
       onRestoreFocus?.({
         originalElement: originalFocusedElement.current,
         containsFocus: containsFocus.current,
         documentContainsFocus: getDocument()?.hasFocus() || false,
       });
 
+      if (nodesToHide) {
+        ariaHidden(nodesToHide, false);
+      }
+
       // De-reference DOM Node to avoid retainment via transpiled closure of _onKeyDown
       originalFocusedElement.current = undefined;
     };
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps -- should only run on first render
-  }, []);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Maybe you can remove this line now? Alternatively you can change it to something like:

```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps -- should only run when nodes to hide change
```